TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -365,5 +365,64 @@ public LockTableStatement CreateLockTableStatement(CodeElementsParser.LockTableS
 
             return new LockTableStatement(tableName, partitionId, mode);
         }
+        public ConnectStatement CreateConnectStatement(CodeElementsParser.ConnectStatementContext context)
+        {
+            ConnectionTarget connectionTarget = null;
+            ConnectionAuthorization connectionAuthorization = null;
+            SyntaxProperty<bool> reset = null;
+            if (context.targetClause() != null)
+            {
+                connectionTarget = CreateConnectionTarget(context.targetClause());
+                if (context.targetClause().authorizationClause() != null)
+                {
+                    connectionAuthorization = CreateConnectionAuthorization(context.targetClause().authorizationClause());
+                }
+
+            }
+            else if (context.authorizationClause() != null)
+            {
+                connectionAuthorization = CreateConnectionAuthorization(context.authorizationClause());
+            }
+
+            else if (context.sqlReset() != null)
+            {
+                reset = new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(context.sqlReset()));
+            }
+
+            return new ConnectStatement(connectionAuthorization, reset, connectionTarget);
+        }
+
+        public ConnectionTarget CreateConnectionTarget(CodeElementsParser.TargetClauseContext context)
+        {
+            SyntaxProperty<string> locationNameLiteral = null;
+            HostVariable locationNameVariable = null;
+            if (context.locationName != null)
+            {
+                locationNameLiteral = new SyntaxProperty<string>(context.locationName.ToString(), ParseTreeUtils.GetFirstToken(context.UserDefinedWord()));
```

Comment:
What is the type of locationName ? Is there a property that gives the text ? I'm not sure about ToString method as it can be overloaded and is generally not suited to return a single property of an object.

Reply:
LocationName is an IToken . There is  ``` context.locationName.Text ``` .
 I change that  ?